nncf/pruning/filter_pruning/global_ranking/LeGR.py

```diff
@@ -0,0 +1,57 @@
+"""
+ Copyright (c) 2020 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+from nncf.pruning.filter_pruning.global_ranking.RL_evolution import EvolutionOptimizer, LeGREvolutionEnv, LeGRPruner
+from nncf.nncf_logger import logger as nncf_logger
+
+
+class LeGR:
+    def __init__(self, pruning_ctrl, target_model, legr_init_args, train_steps=200, generations=400, max_pruning=0.8):
```

Comment:
`max_pruning=0.8` might be way too prohibitive. I'd suggest setting a lower (safer) default.

Reply:
Fixed, also added target pruning rate as default `max_pruning`.